File: Rust/Math/Can-Make-Arithmetic-Progression-From-Sequence.rs
pub struct Solution;

impl Solution {
    pub fn can_make_arithmetic_progression(arr: Vec<i32>) -> bool {
        if arr.len() == 2 {
            return true;
        }

        let mut arr = arr;
        arr.sort();

        let difference = arr[1] - arr[0];
        arr.iter().enumerate().all(|(index, val)| {
            if index <= arr.len() - 2 && arr[index + 1] - *val != difference {
                return false;
            }
            true
        })
    }
}
fn main() {}

File: Javascript/Can-Make-Arithmetic-Progression-From-Sequence.js
/**
 * @param {number[]} arr
 * @return {boolean}
 */
const canMakeArithmeticProgression = (arr) => {
  if (arr.length == 2) return true;

  arr.sort((a, b) => a - b);

  const difference = arr[0] - arr[1];

  for (let i = 1; i < arr.length - 1; i++) {
    if (arr[i] - arr[i + 1] !== difference) return false;
  }

  return true;
};
